react/features/filmstrip/components/web/Thumbnail.js

```diff
@@ -413,6 +417,40 @@ class Thumbnail extends Component<Props, State> {
      */
     componentDidMount() {
         this._onDisplayModeChanged();
+
+
+        // Listen to track streaming status changed event to keep it updated.
+        // TODO: after converting this component to a react function component,
+        // use a custom hook to update local track streaming status.
+        const { _videoTrack, dispatch, _sourceNameSignalingEnabled } = this.props;
+
+        if (_sourceNameSignalingEnabled && _videoTrack && !_videoTrack.local) {
+            _videoTrack.jitsiTrack.on(JitsiTrackEvents.TRACK_STREAMING_STATUS_CHANGED,
+                this.handleTrackStreamingStatusChanged);
+            dispatch(trackStreamingStatusChanged(_videoTrack.jitsiTrack,
+                _videoTrack.jitsiTrack.getTrackStreamingStatus()));
+        }
+
+        if (_videoTrack && !_videoTrack.local) {
+            console.log('trackSettings', _videoTrack.jitsiTrack.track.getSettings())
+        }
+    }
+
+    /**
+     * Remove listeners for track streaming status update.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        const { _videoTrack, dispatch, _sourceNameSignalingEnabled } = this.props;
+
+        if (_sourceNameSignalingEnabled && _videoTrack && !_videoTrack.local) {
+            _videoTrack.jitsiTrack.off(JitsiTrackEvents.TRACK_STREAMING_STATUS_CHANGED,
```

Comment:
track.off also needs to be called on props change?

Reply:
Good question. Do you mean we need to do track.off and track.on when this.props._videoTrack change? If so, what condition triggers this.props._videoTrack change? How to check if this.props._videoTrack is changed?

Reply:
in `componentDidUpdate` just compare the instances:
https://github.com/jitsi/jitsi-meet/pull/11377/files#diff-915778d31824bfed939a12755b3e97ae0addefa8acfb6c45127ae8e914ef55c4R466

Reply:
added.